### PR TITLE
add pragma to params of macros.newProc

### PR DESCRIPTION
**Pull request by loloiccl**

### CODE REVIEW

File: lib/core/macros.nim [hunk 1]
```diff
@@ -1076,20 +1076,24 @@ proc expectKind*(n: NimNode; k: set[NimNodeKind]) {.compileTime.} =
   ## macros that check the AST that is passed to them.
   if n.kind notin k: error("Expected one of " & $k & ", got " & $n.kind, n)
 
-proc newProc*(name = newEmptyNode(); params: openArray[NimNode] = [newEmptyNode()];
-    body: NimNode = newStmtList(), procType = nnkProcDef): NimNode {.compileTime.} =
+proc newProc*(name = newEmptyNode();
+              params: openArray[NimNode] = [newEmptyNode()];
+              body: NimNode = newStmtList();
+              procType = nnkProcDef;
+              pragma: NimNode = newEmptyNode()): NimNode {.compileTime.} =
```
Comment (Araq) [inline lib/core/macros.nim]: It should be ``pragmas``, plural.

**Comment by Araq** [comment]
Apart from my remark, very nice!

**Comment by loloiccl** [comment]
fixed.
thanks for your review!

File: lib/core/macros.nim (hunk 1 above)
Comment (loloiccl) [reply lib/core/macros.nim]: Do you mean new commit or ``pragmas: openArray[NimNode]`` ?
If you mean later, should pragma be nnkIdent?

Comment (Araq) [reply lib/core/macros.nim]: All I'm saying is that parameter's name shall be ``pragmas``.

Comment (loloiccl) [reply lib/core/macros.nim]: Thanks.
Then, the newer commit is fixed one.
Thanks for retriggering CI.

**Comment by loloiccl** [comment]
Failed test is related with my library ``nimly``
There are some problems with it.
https://github.com/loloiccl/nimly/issues/11
https://github.com/loloiccl/nimly/issues/24
Are these related with this failure?

**Comment by Araq** [comment]
@narimiran are we sure nimly works on devel otherwise?

**Comment by narimiran** [comment]
> are we sure nimly works on devel otherwise?

I just checked and it seems to be wonky:
* in 485d5448fa325b74b3636ffd53ede4066c3aad96 it worked
* a commit after (0e6eb7d483fe1b975bd51a4900946130edf506e7) if failed
* a commit after (fdc3f54f9760b6512c5bdc3af6ed034187e70ffe) if was ok again
* then fail again, and ok after it

I'll investigate what is going on, and if I can't reproduce it maybe we should disable it until it is consistent.

(In the mean time, this can be merged)

**Pull request merged**